examples/seq2seq/finetune_trainer.py

```diff
@@ -252,45 +293,53 @@ def main():
         data_args=data_args,
     )
 
+    all_metrics = {}
     # Training
     if training_args.do_train:
+
+        t0 = time.time()
         trainer.train(
             model_path=model_args.model_name_or_path if os.path.isdir(model_args.model_name_or_path) else None
         )
+        metrics = speed_metrics("train", t0, data_args.n_train)
+
         trainer.save_model()
-        # For convenience, we also re-save the tokenizer to the same directory,
-        # so that you can share your model easily on huggingface.co/models =)
+
         if trainer.is_world_process_zero():
+            handle_metrics("train", metrics, training_args.output_dir)
+            all_metrics.update(metrics)
+
+            # For convenience, we also re-save the tokenizer to the same directory,
+            # so that you can share your model easily on huggingface.co/models =)
             trainer.state.save_to_json(os.path.join(training_args.output_dir, "trainer_state.json"))
             tokenizer.save_pretrained(training_args.output_dir)
```

Comment:
Note: `Trainer.save_model` saves the tokenizer with the model (for easy deployment to huggingface.co/models). It does not save the state however so the the line that saves the state is still necessary.

Reply:
note - I only moved the comment to where it belongs since the if branch got unrealted code in it in this PR, and didn't write it.

So you're saying the original comments is inaccurate and should instead say:

```
# Trainer.save_model saves the tokenizer with the model, but not the state. So we do it explicitly
```

if not please suggest what the accurate comment should be. Thanks.

Reply:
That's accurate!

Reply:
Sorry, what I wrote as a suggestion or the original?

i.e. leave the original as is, or replace it with the new comment I wrote above?

Reply:
The new comment you wrote above is accurate.

Reply:
OK, but it's in addition to the existing comments, so what really should be happening is:

```
            # Need to save the state, since Trainer.save_model saves only the tokenizer with the model
            trainer.state.save_to_json(os.path.join(training_args.output_dir, "trainer_state.json"))
            
            # For convenience, we also re-save the tokenizer to the same directory,
            # so that you can share your model easily on huggingface.co/models =)
            tokenizer.save_pretrained(training_args.output_dir)
```

Please confirm. Thank you.

Reply:
Replace the second comment by
```
            # For convenience, this method also saves the tokenizer to the same directory,
            # so that you can share your model easily on huggingface.co/models =)
```

Reply:
what does "this method" refer to?

Reply:
Perhaps "For convenience, we save the tokenizer to the same directory,"?

Reply:
these comments are odd anyway, they should probably be logger comments

Reply:
The more I think about it, the more I believe finetune_trainer should be an app and not an example.

Reply:
I don't understand why this method would be ambiguous when the next line is `trainer.save_model`. We can replace "this method" by `Trainer.save_model` if necessary.

Reply:
I see why this discussion is confusing - please have a look at the HEAD of this PR and not the original. There is no ` trainer.save_model()` where the comments are.